app/controllers/users_controller.rb

```diff
@@ -19,30 +19,31 @@ def user_guardians
     if admin?
       @guardians = current_user.guardians.all
       raise ActiveRecord::RecordNotFound unless @guardians
+
       render json: @guardians, status: :ok
     else
       render json: { error: 'Not Allowed' }, status: :unauthorized
     end
   end
 
   def add_guardian
-    if admin?
-      if User.find_by_email(user_params[:email])
-        render json: { error: 'Email Exist , try a diffrent one' }, status: :not_acceptable
+    return unless admin?
```

Comment:
Again - admin-only actions can be handled by a callback to make the controller more clean. IT also deserves n error message - so what if we have a private method in the Application controller that checks if a user is an admin and returns an error if not? It could be used in the callbacks list for each controller.